File: app/src/main/java/sgtmelon/scriptum/app/view/callback/NoteCallback.java
package sgtmelon.scriptum.app.view.callback;

import androidx.annotation.NonNull;
import sgtmelon.scriptum.app.control.SaveControl;
import sgtmelon.scriptum.app.view.activity.NoteActivity;
import sgtmelon.scriptum.app.view.parent.NoteFragmentParent;
import sgtmelon.scriptum.app.vm.activity.NoteViewModel;

/**
 * Интерфейс общения {@link NoteFragmentParent} и {@link NoteActivity}
 */
public interface NoteCallback {

    void setupFragment(boolean isSave);

    @NonNull
    SaveControl getSaveControl();

    @NonNull
    NoteViewModel getViewModel();

}

File: app/src/main/java/sgtmelon/scriptum/app/screen/callback/NoteCallback.kt
package sgtmelon.scriptum.app.screen.callback

/**
 * Интерфейс общения [NoteViewModel] с [NoteActivity]
 */
interface NoteCallback {

    fun showTextFragment(id: Long, isSave: Boolean)

    fun showRollFragment(id: Long, isSave: Boolean)

    fun onPressBackText(): Boolean

    fun onPressBackRoll(): Boolean

    fun finish()

}